conda/core/path_actions.py

```diff
@@ -1042,3 +1042,34 @@ def target_full_path(self):
     def __str__(self):
         return ('ExtractPackageAction<source_full_path=%r, target_full_path=%r>'
                 % (self.source_full_path, self.target_full_path))
+
+
+class DumpRecordAction(PathAction):
+
+    def __init__(self, target_pkgs_dir, target_extracted_dirname, record):
+        self.target_pkgs_dir = target_pkgs_dir
+        self.target_extracted_dirname = target_extracted_dirname
+        self.record = record
+
+    def verify(self):
+        self._verified = True
+
+    def execute(self):
+
+        meta = join(self.target_full_path, 'info', 'record.json')
+        log.trace("writing record to %s", meta)
+        with open(meta, 'w') as out:
+            out.write(self.record.pretty_json())
+
+    def reverse(self):
+        return
+
+    def cleanup(self):
+        return
+
+    @property
+    def target_full_path(self):
```

Comment:
For consistency with all other PathActions, `target_full_path` should point to the single path that's actually being mutated by that action, and in this case that would be the `record.json` file.

All of the path actions that go with unlink/link have a pretty clean concept of

    target_full_path = target_prefix + target_short_path

Here, I guess it's unclear what the equivalent of `target_prefix` is.  Is it `target_pkgs_dir`, or `package_cache_entry.extracted_package_dir` (which is a full, absolute path also)?

Looking at the code, for now it makes more sense to me to use the second, so we can keep the consistency with the other path actions.  So the init signature should take only two parameters, where the first is just the equivalent of `join(target_pkgs_dir, target_extracted_dirname)`.  Then, in init, also make a attribute `self.target_short_path = 'info/record.json`.  And the code for the `target_full_path` property should be

    return join(self.target_extracted_package_dir, win_path_ok(self.target_short_path))

And we maintain consistency with the rest of the code in the module.  If there's a good reason to break consistency though, then I doubt I'll have an objection.